src/Downloader.ts

```diff
@@ -677,6 +670,49 @@ class Downloader {
     }
   }
 
+  public async checkAndReplaceWeakEtag(etag: string) {
+    if (etag.substr(0, 2) !== 'W/') {
+        return etag;
+    }
+    const strongEtag =  etag.replace('W/', '').replace(/"/g, '');
+    return strongEtag;
+  }
+
+  private async checkStatusAndUploadtoS3(requestOptions: any, handler: any, imageResp: any) {
+    if (imageResp) {
+      requestOptions.headers['If-None-Match'] = await this.checkAndReplaceWeakEtag(imageResp.Metadata.etag);
+    }
+
+    try {
+      const resp = await axios(requestOptions);
+      const etag = resp.headers.etag;
+      const content = await this.getCompressedBody(resp);
+      const compressionWorked = content.length < resp.data.length;
+      if (compressionWorked) {
+        resp.data = content;
+      }
+
+      if (etag) {
+        this.s3.uploadBlob(this.stripHttpFromUrl(requestOptions.url), resp.data, etag);
+      }
+
+      handler(null, {
+        responseHeaders: resp.headers,
+        content: compressionWorked ? content : resp.data,
+      });
```

Comment:
This code seems OK, but is a bit unclear:
* First of all put a clear comment after the axios request to say that most of the time the code won't be executed becaue of the HTTP 304
* Then handle the compression stuff
* Then handle the etag/upload stuf
* Don't understand why you put `content` in `resp.data`

Reply:
Actually you should not treat status HTTP 304 as an error. You should tell Axios with `validateStatus` that 304 is OK and then treat it properly.